### ISSUE TITLE
Media auto-download (sometimes) does not work

### ISSUE THREAD

**Issue by halb9** [signalapp/Signal-Android#10775]
- [X] I have searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I set Media auto-download for WiFi and mobile data
![signal-2021-01-23-141526](https://user-images.githubusercontent.com/18451020/105579259-b7c35f00-5d85-11eb-836c-6b315a1a4513.png)
Most of the time this works. Sometimes it doesn't. Couldn't figure out when exactly. 
When it does not, the image just shows as blurred preview and I have to tap on it to download. 
Contacts are in my address book. This happens on WiFi and mobile data. 
The first log is from when I noticed it. After that I tapped the image to download. And after that I created the second log. 

### Steps to reproduce
- just wait patiently :)

**Actual result:** Media shows as a blurred preview. No full download. 
**Expected result:** Media downloads right away. 

### Device info

**Device:** Xiaomi Mi A1
**Android version:** Android 10 - lineage-17.1-20210119

**Signal version:** 5.3.1

### Link to debug log

https://debuglogs.org/b49d88ae433b14d8b586dcc778436fef55ce092ee1f8cf26f8a766ec5466b004
https://debuglogs.org/7071de17daa7ba50cae6472efde9aa2128d7a0e374dc53f1b0b0256248065031

Thanks!

**Comment by atombyte** [comment]
I have the same issue on my OnePlus 6T with Android 10.

The auto-download worked for me only couple of days after I installed the App. Now I have to tap every media to download it.
Even with all enabled auto-download checkboxes.

Still trying to figure it out, which probably other of the changed settings made this side effect.

**Comment by greyson-signal** [comment]
We recently disabled auto-download while on a call. Do you think that could be it?

**Comment by atombyte** [comment]
Amazing, I just figured it out!

I have had set global lifetime for all messages to 30 days - that 
was the reason, why no media was downloaded automatically.

I switched back to "keep messages forever" and now it works like a charm.

![Screenshot_20210124-103907](https://user-images.githubusercontent.com/35140189/105626680-9455ee00-5e31-11eb-8b6e-c742c0692591.jpg)

**Comment by halb9** [comment]
> We recently disabled auto-download while on a call. Do you think that could be it?

No. I was not on a call at that time. 

> Amazing, I just figured it out!
> 
> I have had set global lifetime for all messages to 30 days - that
> was the reason, why no media was downloaded automatically.
> 
Interesting. I have set my message limit to 30 days as well. 

Yesterday I received a picture in a group chat. **The picture was downloaded!** 
Today the message was blurred. And I had to tap it to "download again". 

![signal-2021-01-24-115259](https://user-images.githubusercontent.com/18451020/105628444-3cbd7f80-5e3d-11eb-9395-cbc2aee3c483.png)
![signal-2021-01-24-115303](https://user-images.githubusercontent.com/18451020/105628447-447d2400-5e3d-11eb-9e93-6a802a338c64.png)

Again like in my first post two debuglogs. First from when I noticed it. Second from after I tapped to download. 
1. https://debuglogs.org/29681cded3c52e720bbc78951ee66ad6c71e111d82d51e44ed1c5c1bea726168
2. https://debuglogs.org/d326207f962eb7fce066d85c4a8a1df9a377bbc3d6f0e6643c4cc12942ccf361

Thanks!

**Comment by halb9** [comment]
Today I received a few pictures in succession while I was looking at the screen. One of it did not load. 
![signal-2021-01-27-102400](https://user-images.githubusercontent.com/18451020/106032946-68f12e80-60d1-11eb-8a67-db1a0e71f025.png)
https://debuglogs.org/d52849e86c73ab1c520c53da24d1f0bcdba62b8be128817ff5d01b5ca62dc4b2
Hope this helps! Thanks!

**Comment by pkorba** [comment]
I have the same problem. It happens randomly on both mobile data and Wi-Fi. I was not on a call or anything - phone was lying idle when I got the message. I have global lifetime for all messages set to 1 year. I also noticed this happening once or twice on one of the previous versions of desktop client but I was busy and then forgot to report that back then. I'll provide desktop debug log if this happens again. Here's debug log from Android client:

https://debuglogs.org/cb2fc69714915d0e36e4b7d1f9abcf4460b043f4fd1c2fb68c5823989084e3a4

![Screenshot_20210220-185701](https://user-images.githubusercontent.com/32198274/108607563-68427400-73c1-11eb-9f55-8d8aee6778c1.png)
![Screenshot_20210220-185732](https://user-images.githubusercontent.com/32198274/108607564-68db0a80-73c1-11eb-9629-ac98948c3ac1.png)
